protocol-units/bridge/contracts/src/AtomicBridgeCounterparty.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.22;
+
+import {Initializable} from "@openzeppelin/contracts/proxy/utils/Initializable.sol";
+import {IAtomicBridgeCounterparty} from "./IAtomicBridgeCounterparty.sol";
+import {AtomicBridgeInitiator} from "./AtomicBridgeInitator.sol";
+
+contract AtomicBridgeCounterparty is IAtomicBridgeCounterparty, Initializable {
+    enum MessageState {
+        PENDING,
+        COMPLETED,
+        REFUNDED
+    }
+
+    struct BridgeTransferDetails {
+        bytes32 initiator; // movement address 
+        address recipient;
+        uint256 amount;
+        bytes32 hashLock;
+        uint256 timeLock;
+        MessageState state; 
+    }
+
+    AtomicBridgeInitiator public atomicBridgeInitiator;
+    mapping(bytes32 => BridgeTransferDetails) public bridgeTransfers; 
+
+    function initialize(address _atomicBridgeInitiator) public initializer {
+        if (_atomicBridgeInitiator == address(0)) revert ZeroAddress();
+        atomicBridgeInitiator = AtomicBridgeInitiator(_atomicBridgeInitiator);
+    }
+
+    function lockBridgeTransferAssets(
+        bytes32 initiator,
+        bytes32 bridgeTransferId,
+        bytes32 hashLock,
+        uint256 timeLock,
+        address recipient,
+        uint256 amount
+    ) external returns (bool) {
+        BridgeTransferDetails storage transfer = bridgeTransfers[bridgeTransferId];
+        if (transfer.recipient != address(0)) revert BridgeTransferInvalid();
+        if (amount == 0) revert ZeroAmount();
+        bridgeTransfers[bridgeTransferId] = BridgeTransferDetails({
+            recipient: recipient,
+            initiator: initiator,
+            amount: amount,
+            hashLock: hashLock,
+            timeLock: block.timestamp + timeLock,
+            state: MessageState.PENDING 
+        });
+
+        emit BridgeTransferAssetsLocked(bridgeTransferId, recipient, amount, hashLock, timeLock);
+
+        return true;
```

Comment:
Do you mean the case where the same address might be swapping at the same time, as that recipient address is the key for the balances mapping?

What's the suggested change here? 

In the case of the same recipient address, we could use `bridgeTransferIds` to store the locked WETH instead of address, that's probably better in any case so I will add this.

Reply:
Makse sense! Ok

Reply:
The check now happens inside the `withdrawETH` function 
`if (poolBalance =< amount) revert ZeroAmount();`